.github/workflows/expire.yml

```diff
@@ -110,15 +117,10 @@ jobs:
                   body: `**\`Access on Demand\`** - request closed and branch deleted due to inactivity (no new commit) for at least ${{ inputs.expiry_hours }} hours`,
                 });
 
-                core.info(`Deleting branch ${ref}`);
-
-                // Delete branch.
-                await github.rest.git.deleteRef({
-                  owner: repoOwner,
-                  repo: repoName,
-                  ref: `heads/${ref}`,
-                });
+                expiredPulls.push({pullNumber: pull['number'], pullBranch: pull['head']['ref']})
               } else {
                 core.info(`Skipping #${pull['number']} (${pull['title']}) - not an AOD request`);
               }
-            });
+            }));
+
+            return expiredPulls;
```

Comment:
Is the plan to for each expired pulls to call cleanup workflow?

Reply:
yes, we can use github matrix to run them parallelly, per Github doc, a matrix will generate a maximum of 256 jobs per workflow run.